src/lib/y2storage/proposal/autoinst_space_maker.rb

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/disk"
+
+module Y2Storage
+  module Proposal
+    # Class to provide free space during the AutoYaST proposal by deleting
+    # partitions and partition tables according to the information in the
+    # AutoYaST profile.
+    class AutoinstSpaceMaker
```

Comment:
another idea. Can be common SpaceMaker used just with settings tuned according to autoyast profile? I would like to see sharing functionality :)